src/prettier-comments/language-js/comments.js

```diff
@@ -257,21 +257,51 @@ function handleIfStatementComments(
     precedingNode === enclosingNode.trueBody &&
     followingNode === enclosingNode.falseBody
   ) {
-    if (precedingNode.type === "ExpressionStatement") {
-      addTrailingComment(precedingNode, comment);
+    if (followingNode.type === "Block") {
```

Comment:
In slang we have way more control over comments and inspired by Prettier's decision of no having comments between the `trueBody` and `falseBody` we are gonna put it at the beginning of the `falseBody`. This also removes a lot to edge case logic in `ExpressionStatement.js` and `IfStatement.js`